scipy/fft/tests/test_basic.py

```diff
@@ -40,26 +40,15 @@ def fft1(x):
     phase = np.arange(L).reshape(-1, 1) * phase
     return np.sum(x*np.exp(phase), axis=1)
 
-
-class TestFFTShift:
-
-    def test_fft_n(self, xp):
-        x = xp.asarray([1, 2, 3], dtype=xp.complex128)
-        if xp.__name__ == 'torch':
-            assert_raises(RuntimeError, fft.fft, x, 0)
-        else:
-            assert_raises(ValueError, fft.fft, x, 0)
-
-
 class TestFFT1D:
 
     def test_identity(self, xp):
         maxlen = 512
         x = xp.asarray(random(maxlen) + 1j*random(maxlen))
         xr = xp.asarray(random(maxlen))
         for i in range(1, maxlen):
-            xp_assert_close(fft.ifft(fft.fft(x[0:i])), x[0:i], rtol=1e-9, atol=0)
-            xp_assert_close(fft.irfft(fft.rfft(xr[0:i]), i), xr[0:i], rtol=1e-9, atol=0)
+            xp_assert_close(fft.ifft(fft.fft(x[0:i])), x[0:i], rtol=1e-7, atol=0)
+            xp_assert_close(fft.irfft(fft.rfft(xr[0:i]), i), xr[0:i], rtol=1e-7, atol=0)
```

Comment:
this test takes ~80 seconds in CI, presumably because it is when JAX is initialised. Hence, the bump on timeout to `120` in the workflow. That isn't ideal, but not a big deal as the array API workflow is much shorter than others at the minute. Maybe it would become a problem down the line.

Reply:
The reason this takes so long is the same reason the array API test suite times out for JAX, as reported here: https://github.com/data-apis/array-api-tests/issues/197

Namely, JAX is pretty slow when eagerly dispatched, and you're dispatching 1024 separate operations here, each of which needs to be separately compiled because the input shapes are changing.

Reply:
Right, thanks for the explanation, I see. In which case, over 1000 calls seems very excessive.

Reply:
It's excessive even with `numpy`, and it's going to be slower for libraries other than JAX as well. It takes:
```
0.24s call     build-install/lib/python3.11/site-packages/scipy/fft/tests/test_basic.py::TestFFT1D::test_identity[array_api_strict]
0.16s call     build-install/lib/python3.11/site-packages/scipy/fft/tests/test_basic.py::TestFFT1D::test_identity[numpy]
```

I'd suggest keeping the `1e-9` as a numpy-only test (it may be useful to exercise `pocketfft`, and cutting down on this test substantially for other libraries (just spot-checking with some powers of 2 and some prime numbers will be fine).

Reply:
Okay, down from `512*8` ops to `9*8` for non-np now.

```python
    def test_identity(self, xp):
        maxlen = 512
        x = xp.asarray(random(maxlen) + 1j*random(maxlen))
        xr = xp.asarray(random(maxlen))
        # Check some powers of 2 and some primes
        for i in [1, 2, 16, 128, 512, 53, 149, 281, 397]:
            xp_assert_close(fft.ifft(fft.fft(x[0:i])), x[0:i], rtol=1e-9, atol=0)
            xp_assert_close(fft.irfft(fft.rfft(xr[0:i]), i), xr[0:i], rtol=1e-9, atol=0)

    @skip_xp_backends(np_only=True, reasons=['significant overhead for some backends'])
    def test_identity_extensive(self, xp):
        maxlen = 512
        x = xp.asarray(random(maxlen) + 1j*random(maxlen))
        xr = xp.asarray(random(maxlen))
        for i in range(1, maxlen):
            xp_assert_close(fft.ifft(fft.fft(x[0:i])), x[0:i], rtol=1e-9, atol=0)
            xp_assert_close(fft.irfft(fft.rfft(xr[0:i]), i), xr[0:i], rtol=1e-9, atol=0)
```